### DISCUSSION TITLE
Iped external open - name code

### DISCUSSION THREAD

**Discussion by gisms** [sepinf-inc/IPED#1406] [Q&A] [answered]
Hi everyone!!

when we open a file externally by iped it creates a new file in dir indextemp, with a code in the name, example: iped7519723097774681305.pdf

is this code created randomly?
is it possible to trace the original file by this code?

**Comment by lfcnassif** [comment, marked as answer]
> is this code created randomly?

Yes

> is it possible to trace the original file by this code?

Unfortunately not.

**Comment by wladimirleite** [comment]
> Is it possible to trace the original file by this code?

As @lfcnassif replied, it is not possible to trace back the original file from the temporary file name, but what is the actual use case? Why do you need/want to do that?

**Reply by lfcnassif** [reply]
If it is important, maybe we could change to use the file hash when available. At least in `fastmode` profile, it is not. There are also cases where artifacts don't have a hash, see #946

**Reply by gisms** [reply]
> > Is it possible to trace the original file by this code?
> 
> As @lfcnassif replied, it is not possible to trace back the original file from the temporary file name, but what is the actual use case? Why do you need/want to do that?


no, it's not a necessity, just understanding how it works. thank u all

**Reply by paulobreim** [reply]
When you open a file (like pdf) by double clicking, the file name is like this.

You can save the original file by right clicking on the line where you selected file to view.
